pageserver/src/tenant.rs

```diff
@@ -1296,11 +1966,15 @@ pub mod harness {
                     .to_string_lossy()
                     .parse()?;
 
-                let timeline_metadata = load_metadata(self.conf, timeline_id, self.tenant_id)?;
-                timelines_to_load.insert(timeline_id, timeline_metadata);
-            }
-            tenant.init_attach_timelines(timelines_to_load)?;
+                let timeline = tenant.load_local_timeline(timeline_id)?;
 
+                // Insert it to the hash map
+                tenant
+                    .timelines
+                    .lock()
+                    .unwrap()
+                    .insert(timeline_id, timeline);
+            }
```

Comment:
Not your PR to blame, but it did not make the issue easier: we need to split this file eventually.
I think after async and status additions, it might be a good time to do so: `tenant_tasks.rs` and `tenant_mgr.rs` are quite empty now.